matchzoo/datapack.py

```diff
@@ -0,0 +1,114 @@
+"""Matchzoo DataPack, pair-wise tuple (feature) and context as input."""
+
+import typing
+from pathlib import Path
+
+import dill
+import pandas as pd
+
+
+class DataPack(pd.DataFrame):
```

Comment:
Using `pandas.DataFrame` is great but inheritance is a good idea here. Use delegation instead.

Reply:
define a proxy class?

Reply:
@bwanglzu Kinda of. It's more than a proxy class because it has its own special behaviors.